utils/saasherder.py

```diff
@@ -122,18 +122,49 @@ def _get_file_contents(self, options):
         path = options['path']
         ref = options['ref']
         github = options['github']
+        html_url = os.path.join(url, 'blob', ref, path)
+        content = None
         if 'github' in url:
             repo_name = url.rstrip("/").replace('https://github.com/', '')
             repo = github.get_repo(repo_name)
             f = repo.get_contents(path, ref)
-            return f.decoded_content, f.html_url
+            content = f.decoded_content
         elif 'gitlab' in url:
             if not self.gitlab:
                 raise Exception('gitlab is not initialized')
             project = self.gitlab.get_project(url)
             f = project.files.get(file_path=path.lstrip('/'), ref=ref)
-            html_url = os.path.join(url, 'blob', ref, path)
-            return f.decode(), html_url
+            content = f.decode()
+
+        return yaml.safe_load(content), html_url
+
+    @retry()
+    def _get_directory_contents(self, options):
+        url = options['url']
+        path = options['path']
+        ref = options['ref']
+        github = options['github']
+        html_url = os.path.join(url, 'tree', ref, path)
+        resources = []
+        if 'github' in url:
+            repo_name = url.rstrip("/").replace('https://github.com/', '')
+            repo = github.get_repo(repo_name)
+            for f in repo.get_contents(path):
+                file_path = os.path.join(path, f.name)
+                file_contents = repo.get_contents(file_path, ref)
+                resource = yaml.safe_load(file_contents.decoded_content)
+                resources.append(resource)
+        elif 'gitlab' in url:
+            if not self.gitlab:
+                raise Exception('gitlab is not initialized')
+            project = self.gitlab.get_project(url)
+            for f in project.repository_tree(path=path.lstrip('/'), ref=ref):
+                file_contents = \
+                    project.files.get(file_path=f['path'], ref=ref)
+                resource = yaml.safe_load(file_contents.decode())
+                resources.append(resource)
```

Comment:
Why don't we process them? The use case here is for non-templates? Won't the Deployment be in the same directory?

Reply:
the intention is not to process or apply any templating for such files.

in order to template, the resources should be in an openshift template or a helm chart, or anything that has a defined templating method. for the `directory` provider, just simple plain resources.

makes sense?

Reply:
Meaning this is something to be specified in addition to the `openshift-template` provider, not instead of. Got it.

Reply:
https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/6177 for more info